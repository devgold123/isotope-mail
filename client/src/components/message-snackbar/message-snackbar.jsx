import React, {Component} from 'react';
import {connect} from 'react-redux';
import Snackbar from '../snackbar/snackbar';
import {translate} from 'react-i18next';

class MessageSnackbar extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    const {outbox, t} = this.props;
    const message = outbox === null ? '' :
      t('messageSnackbar.sendingMessage', {progress: outbox.progress * 100});
    return (
      <Snackbar show={outbox !== null} alignStart={true} message={message}/>
    );
  }
}

const mapStateToProps = state => ({
  outbox: state.application.outbox
});

export default connect(mapStateToProps)(translate()(MessageSnackbar));
