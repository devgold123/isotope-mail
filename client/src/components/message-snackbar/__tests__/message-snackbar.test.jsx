import React from 'react';
import {shallow} from 'enzyme/build/index';
import {MessageSnackbar} from '../message-snackbar';

describe('Snackbar component test suite', () => {
  describe('Snapshot render', () => {
    test('Renders hidden', () => {
      // Given
      const outbox = null;

      // When
      const messageSnackbar = shallow(<MessageSnackbar outbox={outbox}/>);

      // Then
      expect(messageSnackbar).toMatchSnapshot();
    });
    test('Renders with progress', () => {
      // Given
      const t = jest.fn((messageKey, {progress}) => `${messageKey} ${progress}`);
      const outbox = {sent: false, progress: 1337};

      // When
      const messageSnackbar = shallow(<MessageSnackbar outbox={outbox} t={t}/>);

      // Then
      expect(messageSnackbar).toMatchSnapshot();
      expect(t).toHaveBeenCalledTimes(1);
    });
    test('Renders with sent status', () => {
      // Given
      const t = jest.fn(messageKey => messageKey);
      const outbox = {sent: true, progress: 1337};

      // When
      const messageSnackbar = shallow(<MessageSnackbar outbox={outbox} t={t}/>);

      // Then
      expect(messageSnackbar).toMatchSnapshot();
      expect(t).toHaveBeenCalledTimes(1);
    });
  });
});
