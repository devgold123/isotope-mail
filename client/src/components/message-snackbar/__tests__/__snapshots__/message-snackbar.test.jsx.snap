// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageSnackbar component test suite Snapshot render Renders hidden 1`] = `
<Snackbar
  alignStart={true}
  buttonLabel=""
  message=""
  show={false}
/>
`;

exports[`MessageSnackbar component test suite Snapshot render Renders with progress 1`] = `
<Snackbar
  alignStart={true}
  buttonLabel=""
  message="messageSnackbar.sendingMessage 133700"
  show={true}
/>
`;

exports[`MessageSnackbar component test suite Snapshot render Renders with sent status 1`] = `
<Snackbar
  alignStart={true}
  buttonLabel=""
  message="messageSnackbar.messageSent"
  show={true}
/>
`;
